src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -57,50 +56,56 @@ public ContributionInfo(Docset docset, GitHubUserCache gitHubUserCache)
-
-                    if (user != null && !excludes.Contains(user.Login) && logins.Add(user.Login))
-                        contributors.Add(user.ToContributor());
                 }
             }
 
-            if (!string.IsNullOrEmpty(authorName))
+            var author = await GetAuthor();
+            if (author != null)
             {
-                if (resolveGitHubUsers)
-                {
-                    // Remove author from contributors if author name is specified
-                    var (error, author) = await _gitHubUserCache.GetByLogin(authorName);
-                    if (error != null)
-                        errors.Add(error);
+                contributors.RemoveAll(c => c.Id == author.Id);
+            }
 
-                    if (excludes.Contains(authorName))
-                        author = null;
-                    if (author != null)
-                        contributors.RemoveAll(c => c.Id == author.Id.ToString());
+            return (errors, author, contributors, updatedDateTime);
 
-                    return (errors, author?.ToContributor(), contributors, updatedDateTime);
+            async Task<Contributor> GetContributor(GitCommit commit)
+            {
+                if (!resolveGitHubUsers)
+                {
+                    return new Contributor { DisplayName = commit.AuthorName, Id = commit.AuthorEmail };
                 }
 
-                return (errors, new Contributor { DisplayName = authorName }, contributors, updatedDateTime);
+                var (error, user) = await _gitHubUserCache.GetByCommit(commit.AuthorEmail, gitHubOwner, gitHubRepoName, commit.Sha);
+                if (error != null)
+                    errors.Add(error);
+
+                return user.ToContributor();
             }
 
-            // When author name is not specified, last contributor is author
-            if (contributors.Count > 0)
+            async Task<Contributor> GetAuthor()
             {
-                var author = contributors[contributors.Count - 1];
-                contributors.RemoveAt(contributors.Count - 1);
-                return (errors, author, contributors, updatedDateTime);
+                if (!string.IsNullOrEmpty(authorName))
+                {
+                    if (resolveGitHubUsers && !excludes.Contains(authorName))
+                    {
+                        // Remove author from contributors if author name is specified
+                        var (error, result) = await _gitHubUserCache.GetByLogin(authorName);
+                        if (error != null)
+                            errors.Add(error);
+                        return result?.ToContributor();
+                    }
+                }
+                else if (contributors.Count > 0)
+                {
+                    // When author name is not specified, last contributor is author
+                    return await GetContributor(commits[commits.Count - 1]);
```

Comment:
>commits[commits.Count - 1] [](start = 48, length = 26)

If the last contributor is null, should use the second last contributor as author